StarryExpanse/RivenGameState.h

```diff
@@ -16,4 +20,16 @@ class ARivenGameState : public AGameStateBase {
 
   UPROPERTY(BlueprintReadOnly)
   URivenSaveGame *Instantaneous_SaveGame;
+
+  UPROPERTY()
+  TMap<uint32, TWeakObjectPtr<AActor>> SubscribedToSavegame;
```

Comment:
What value is a map here? Why not just use a TArray?

Reply:
Well, every time we want to remove an item we'd have to iterate the array via linear search. So when you leave a loaded area it'd be querying a quite long list of N, N-1, N-2 etc. elements and that doesn't seem ideal. I wanted a `TSet` originally but I wasn't sure how well UE4 would be able to hash/identify Actors and this seemed easy and transparent.